src/libraries/System.Collections/src/System/Collections/Generic/PriorityQueue.cs

```diff
@@ -247,32 +236,28 @@ public TElement EnqueueDequeue(TElement element, TPriority priority)
         {
             if (_size != 0)
             {
-                (TElement Element, TPriority Priority) root = _nodes[RootIndex];
+                ref (TElement Element, TPriority Priority) root = ref _nodes[0];
```

Comment:
Nit: can you name the locals that are refs to have a "Ref" suffix, e.g. "rootRef"?  Otherwise it's easy to accidentally write over something you weren't expecting.